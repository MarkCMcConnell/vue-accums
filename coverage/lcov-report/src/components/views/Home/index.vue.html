<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/views/Home/index.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/views/Home</a> index.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;section class="container"&gt;
    &lt;BaseModal v-if="this.$store.state.modal.showModal"&gt;
      &lt;h2 slot="modalTitle"&gt;Get Your Yearly Check-Up!&lt;/h2&gt;
      &lt;p
        slot="modalBody"
      &gt;It's that time of year! Call your doctor and get scheduled for a check-up today!&lt;/p&gt;
      &lt;span slot="modalButton"&gt;Proceed&lt;/span&gt;
    &lt;/BaseModal&gt;
    &lt;h1&gt;Home&lt;/h1&gt;
    &lt;CoveragesDashboard
      :coverageData="this.coverages"
      :isDashboardFormat="coverageDashboard"
      :yearRange="1"
    /&gt;
  &lt;/section&gt;
&lt;/template&gt;
&nbsp;
&lt;script&gt;
// General UI Components
import BaseModal from '../../ui/BaseModal.vue'
// Layout components
import CoveragesDashboard from '../../layout/CoveragesDashboard.vue'
&nbsp;
export default {
  components: {
    BaseModal,
    CoveragesDashboard
  },
  data () {
    return {
      member: this.$store.state.memberData[0],
      coverages: this.$store.state.memberData[0].Coverages,
      coverageDashboard: true
    }
  }
}
&lt;/script&gt;
&nbsp;
&lt;style scoped&gt;
.container {
  position: relative;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 16 2019 15:24:25 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
