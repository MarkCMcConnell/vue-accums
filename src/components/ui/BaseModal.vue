<template>
    <div class="modalBackdrop">
        <div class="modal-container">
            <header class="modal-header">
                    <slot name="modalTitle"></slot>
            </header>

            <section  class="modal-body">
                <slot name="modalBody"/>
            </section>
            <footer class="modal-footer">
                <slot name="modalFooter" />
                <button
                    class="modal-button btn-primary"
                    @click.prevent="HIDE_MODAL"
                >
                    <slot name="modalButton">Close</slot>
                </button>
            </footer>
        </div>
    </div>
</template>

<script>
import { mapMutations } from 'vuex'

export default {
    methods: {
        ...mapMutations([
            'HIDE_MODAL'
        ])
    }
}
</script>

<style scoped>
    .modalBackdrop {
        position: fixed;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        background-color: rgba(0, 0, 0, 0.5);
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .modal-container {
        width: 50%;
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate( -50%, -50%);
        background-color: #fff;
    }
</style>
