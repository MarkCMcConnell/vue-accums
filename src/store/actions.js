// Root Actions
const actions = {
    CHANGE_VIEW ({ commit }, payload) {
        commit('changeView', payload)
    },
}

export default {
    actions
}