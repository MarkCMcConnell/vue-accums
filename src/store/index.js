import Vue from 'vue'
import Vuex from 'vuex'
// Import mock data
import memberData from '../../data/SampleElig.js'
import claims from '../../data/SampleClaims.js'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    memberData: memberData,
    claims: claims,
    currentView: 'Home'
  },
  mutations: {
    CHANGE_VIEW (state, payload) {
      state.currentView = payload
    }
  },
  actions: {
    CHANGE_VIEW ({ commit }, payload) {
      commit('changeView', payload)
    }
  },
  getters: {

  }
})
