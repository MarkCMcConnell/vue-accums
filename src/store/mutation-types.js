// Modal types
export const HIDE_MODAL = 'HIDE_MODAL'

// Navbar types
export const CHANGE_VIEW = 'CHANGE_VIEW'
