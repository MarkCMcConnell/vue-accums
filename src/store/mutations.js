// Root Mutations
const mutations = {
    CHANGE_VIEW (state, payload) {
        state.currentView = payload
    },
}

export default {
    mutations
}