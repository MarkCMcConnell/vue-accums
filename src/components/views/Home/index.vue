<template>
    <section class="container">
        <h1>Home</h1>
        <BaseModal v-if="this.$store.state.modal.showModal">
            <h2 slot="modalTitle">Get Your Yearly Check-Up!</h2>
            <p slot="modalBody">It's that time of year!  Call your doctor and get scheduled for a check-up today!</p>
            <span slot="modalButton">Proceed</span>
        </BaseModal>
    </section>
</template>

<script>
// General UI Components
import BaseModal from '../../ui/BaseModal.vue'

export default {
  components: {
    BaseModal
  }
}
</script>

<style scoped>
.container {
  position: relative;
}
</style>
