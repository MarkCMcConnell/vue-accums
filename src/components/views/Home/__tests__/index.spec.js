/* eslint-disable */
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import store from '@/store'

import Home from '@/components/views/Home'
import BaseModal from '@/components/ui/BaseModal.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Home.vue', () => {
  let mockStore
  let wrapper
  beforeEach(() => {
    mockStore = store
    wrapper = mount(Home, {
      store: mockStore,
      localVue,
      stubs: {
        BaseModal: '<div class="stubbed modalBackdrop" />'
      }
    })
  })

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has a header tag', () => {
    expect(wrapper.contains('h1')).toBe(true)
  })
  
  it('displays a modal on first load', () => {
    expect(wrapper.contains('.modalBackdrop')).toBe(true)
  })
})
