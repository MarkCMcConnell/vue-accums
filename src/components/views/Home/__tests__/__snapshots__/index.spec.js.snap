// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders the component 1`] = `
<section class="container">
  <h1>Home</h1>
  <div class="stubbed modalBackdrop"></div>
</section>
`;
